instrumentation/okhttp/okhttp-2.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/okhttp/v2_2/OkHttp2Test.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.okhttp.v2_2;
+
+import com.squareup.okhttp.Callback;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import com.squareup.okhttp.internal.http.HttpMethod;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpClientTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestOptions;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class OkHttp2Test extends AbstractHttpClientTest<Request> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forAgent();
+
+  private static final OkHttpClient client = new OkHttpClient();
+  private static final OkHttpClient clientWithReadTimeout = new OkHttpClient();
+
+  @BeforeAll
+  void setupSpec() {
+    client.setConnectTimeout(CONNECTION_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+    clientWithReadTimeout.setConnectTimeout(CONNECTION_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+    clientWithReadTimeout.setReadTimeout(READ_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+  }
+
+  @Override
+  public Request buildRequest(String method, URI uri, Map<String, String> headers) {
+    try {
+      RequestBody body =
+          HttpMethod.requiresRequestBody(method)
+              ? RequestBody.create(MediaType.parse("text/plain"), "")
+              : null;
+      Request.Builder request = new Request.Builder().url(uri.toURL()).method(method, body);
+      headers.forEach(request::header);
+      return request.build();
+    } catch (MalformedURLException e) {
+      throw new AssertionError("Invalid URL: " + uri, e);
+    }
```

Comment:
what do you think of adding Exception to the throws clause for `buildRequest` instead?

Reply:
Good idea 👍